examples/server/server.cpp

```diff
@@ -3220,15 +3220,20 @@ int main(int argc, char ** argv) {
 
     // if a custom chat template is not supplied, we will use the one that comes with the model (if any)
     if (params.chat_template.empty()) {
-        if (!ctx_server.validate_model_chat_template()) {
+        if (!params.input_prefix.empty() || !params.input_suffix.empty()) {
```

Comment:
I think here we can refactor a bit:
- if both `chat_template`, `suffix/prefix` are set, we throw an error says "only template or suffix/prefix can be set, but not both"
- if `chat_template` is set, we use it
- if `suffix/prefix` are set, we use it and discard `chat_template`
- if none is specified, we use model's default template

In either cases, it would be nice to output a test formatted chat here

Reply:
> * if both `chat_template`, `suffix/prefix` are set, we throw an error says "only template or suffix/prefix can be set, but not both"

I think that prioritizing chat template and ignoring prefix/suffix is enough - and we can replicate the same in UI by hiding or dimming prefix/suffix input fields if "custom" is not selected.

Again, I'm not sure about leaving this in a weird noname state and only using prefix/suffix to define that custom template is used. Clearly stating what it is would be better, I think.

Reply:
For the UI, I think we can go with something simple for now: I'd suggest simply have a checkbox says "use custom chat template" and if it's checked, we show prefix/suffix input. No dropdown is needed.

Let's skip the ability to select named templates like "llama3", "mistral", etc for now, because it's difficult to maintain 2 lists of template, one in `llama.cpp` and another in `index.html`